packages/eslint-plugin-choreography-ts/src/no-outside-choreographic-operator.ts

```diff
@@ -34,55 +40,106 @@ const noOutsideOperatorRule: TSESLint.RuleModule<MessageIDs, []> = {
         ) {
-          curr = curr.parent;
-        }
-        const params = (curr as TSESTree.ArrowFunctionExpression).params;
-        const param = params[0];
-        // if the parameter list isn't empty
-        if (param) {
-          // if first parameter is a dependencies object
-          if (param.type === AST_NODE_TYPES.ObjectPattern) {
-            const properties = param.properties;
-            // if the dependencies object isn't empty
-            if (properties.length > 0) {
-              // `match` tracks if the operator is found
-              let match = false;
-              // `propertyRange` tracks the location range of the last encountered
-              // operator in the dependencies parameter `{locally, colocally, ...}` so we
-              // know where to place the missing operator
-              let propertyRange: [number, number];
-              // find matching operator in the dependencies parameter
-              properties.forEach((prop) => {
-                if (prop.type === AST_NODE_TYPES.Property) {
-                  // Store the range of the current operator
-                  propertyRange = prop.range;
-                  if (prop.key.type === AST_NODE_TYPES.Identifier) {
-                    // This is more readable without using "&&"
-                    if (prop.key.name === operator) {
-                      match = true;
-                    }
+          // Get the name of the nested operator
+          const operator = (node.callee as TSESTree.Identifier).name;
+          let curr: TSESTree.Node = node;
+          // Iterate up through the AST until the ancestor ArrowFunctionExpression node
+          // containing the choreography's parameters is found, which is stored in `curr`
+          while (
+            curr.type !== AST_NODE_TYPES.ArrowFunctionExpression &&
+            curr.type !== AST_NODE_TYPES.FunctionExpression &&
+            curr.parent
+          ) {
+            curr = curr.parent;
+          }
+          const params = (
+            curr as
+              | TSESTree.ArrowFunctionExpression
+              | TSESTree.FunctionExpression
+          ).params;
+          const param = params[0];
+          // if the parameter list isn't empty
+          if (param) {
+            // if first parameter is a dependencies object
+            if (param.type === AST_NODE_TYPES.ObjectPattern) {
+              const properties = param.properties;
+              // if the dependencies object isn't empty
+              if (properties.length > 0) {
+                // `propertyRange` tracks the location range of the last encountered
+                // operator in the dependencies parameter `{locally, colocally, ...}` so we
+                // know where to place the missing operator
+                let propertyRange: [number, number];
+                // find matching operator in the dependencies parameter
+                const match = properties.find((prop) => {
```

Comment:
Having a side effect (`propertyRange = prop.range;`) inside `find` is not ideal. Can't you make your find to return the range if there is one and `undefined` otherwise?

Reply:
Hmm, maybe. I'll take a look into it.

Reply:
Actually, unfortunately it seems like I can't since `Array,prototype.find` returns the first alement for which _any_ non-null value is returned, meaning that even if I do return the range in the callback, `find` will return the element for which the range is returned. Would reverting back to using `forEach` be better in this case?

Reply:
@ImpregnableProgrammer Does `findLast` help here? I might be misunderstanding the code, but if it's difficult to use a variant of `find` I'd say just use `forEach` (or `for of`)